src/ASM/ASMs2Dmx.C

```diff
@@ -59,12 +59,21 @@ Go::SplineSurface* ASMs2Dmx::getBasis (int basis) const
 
 Go::SplineCurve* ASMs2Dmx::getBoundary (int dir, int basis)
 {
-  if (dir < -2 || dir == 0 || dir > 2 || basis < 1 || basis > (int)m_basis.size())
+  if (dir < -2 || dir == 0 || dir > 2)
+    return nullptr;
+  else if (basis < 1 || basis > (int)m_basis.size())
     return nullptr;
 
   int iedge = dir > 0 ? dir : 3*dir+6;
-
-  return m_basis[basis-1]->edgeCurve(iedge);
+  if (basis > 1)
+    return m_basis[basis-1]->edgeCurve(iedge);
+  else if (!bou[iedge])
+    // Store the pointer to the first basis' edge in bou, to avoid memory leak
+    // since it is dynamically allocated (will be deallocated by parent class).
+    // If invoked with basis > 1 (less likely), we still leak though..
```

Comment:
that certainly happens in stokes applications. but it's an improvement of course.

Reply:
yes, but only when inhomo dirichlet on pressure (or when different basis for x and y-direction).
Ideally, it should have been handled inside the SplineSurface class (like for the trivariate class).